backend/integration_tests/src/client/user.rs

```diff
@@ -91,6 +97,47 @@ pub mod happy_path {
     }
 
     pub fn initial_state_v2(env: &StateMachine, sender: &User) -> user_canister::initial_state_v2::SuccessResult {
+        fn convert_response(result: SuccessCachedResult) -> user_canister::initial_state_v2::SuccessResult {
+            user_canister::initial_state_v2::SuccessResult {
+                timestamp: result.timestamp,
+                direct_chats: result.direct_chats,
+                group_chats: convert_groups(result.cached_group_chat_summaries, result.group_chats_added),
+                avatar_id: result.avatar_id,
+                blocked_users: result.blocked_users,
+                pinned_chats: result.pinned_chats,
+                user_canister_wasm_version: result.user_canister_wasm_version,
+            }
+        }
+
+        fn convert_groups(
+            cached: Vec<GroupChatSummary>,
+            added: Vec<user_canister::GroupChatSummary>,
+        ) -> Vec<user_canister::GroupChatSummary> {
+            let mut result: Vec<_> = cached.into_iter().map(convert_group_chat_summary).collect();
```

Comment:
Or even this?
`cached.into_iter().map(convert_group_chat_summary).chain(added.into_iter()).collect()`

Reply:
I'll make the change in a later PR